### PR TITLE
Add committee/agency organizations and appointee persons

### PR DESCRIPTION
**Pull request by patcon**
Not ready yet, but just wanted to put it on your radar in case anything I'm doing looks unorthodox. I'm new :)

### CODE REVIEW

File: ca_on_toronto/people.py [hunk 1]
```diff
@@ -46,9 +150,19 @@ def scrape_councilor(self, page, h1, url):
         if address:
             p.add_contact('address', address, 'legislature')
 
-        return p
+        yield p
 
     def scrape_mayor(self, url):
+        # TODO: Fix mayor scraper
```
Comment (patcon) [inline ca_on_toronto/people.py]: This is another issue to fix, but it's just a boring task, so I'm procrastinating and faking a fix for now :)

**Comment by patcon** [comment]
Any thoughts on what might be the reason why my scraper runs when I run `pupa update ca_on_toronto` in the actual toronto dir, but fails with the below error otherwise?

    23:29:38 CRITICAL pupa: no scrapers defined on jurisdiction

**Comment by jpmckinney** [comment]
The `TorontoPersonScraper` shouldn't be scraping everything at once.

Anyway, the reason it can't find scrapers is probably linked to being a subclass of CanadianJurisdiction, which does [this](https://github.com/opencivicdata/scrapers-ca/blob/master/utils.py#L269). Basically, it looks for files called `people.py` and `bills.py` with appropriate class names defined in each (and this code can be changed to look for other Pupa-supported types of scrapers) and it then tells Pupa that it has a Person and Bill scraper.

**Comment by jpmckinney** [comment]
And since you override `__init__`, it no longer tells Pupa about its scrapers via its `self.scrapers` property.

**Comment by patcon** [comment]
Sorry, I'm unfamiliar with Python, so I unfortunately need more guidance to correct things, if you wouldn't mind :)

> The TorontoPersonScraper shouldn't be scraping everything at once.

My understanding was that it was supposed to scrape orgs, persons, and memberships, which is what it's doing, no? Am I misunderstanding your comment?

https://opencivicdata.readthedocs.org/en/latest/scrape/people.html#special-notes-about-people-scrapers

I think it's been acting wonky since before I added __init__ today, and I thought this approach inherited and didn't override. You're saying that's not the case?

Also, I just realized that the error and directory issue above was the result of my local edit to speed up testing. Reverting this solved all errors.

```diff
diff --git a/utils.py b/utils.py
index 52d6b14..31bd430 100644
--- a/utils.py
+++ b/utils.py
@@ -268,7 +268,7 @@ class CanadianJurisdiction(Jurisdiction):
 
     def __init__(self):
         super(CanadianJurisdiction, self).__init__()
-        for module, name in (('people', 'Person'), ('bills', 'Bill')):
+        for module, name in (('people', 'Person'), ):
             try:
                 class_name = self.__class__.__name__ + name + 'Scraper'
                 self.scrapers[module] = getattr(__import__(self.__module__ + '.' + module, fromlist=[class_name]), class_name)
```

Anyhow, it sounded like there were some things you were hoping I could fix, but I'm unsure what those are.

I'll spend some more time checking through the data, but in general, I'm happy with the approach unless I'm informed otherwise, so let me know :)

**Comment by jpmckinney** [comment]
Yeah, nevermind - your inheritance was fine :)

I haven't read everything carefully, but it looks like you're pulling data from multiple independent sources in a single scrape. This is what causes https://github.com/opencivicdata/pupa/issues/215 Here's how Pupa expects things to work:

1. You scrape people from councillor pages in one scraper (anything defined in the jurisdiction's `scrapers`). Pupa reconciles those people with people already saved to the DB.
2. You scrape committee memberships from committee pages in another scraper. Here, you can either:
  * add memberships to a person: in which case you must use [add_membership](https://github.com/opencivicdata/pupa/blob/master/pupa/scrape/popolo.py#L97) and create/scrape and yield the people and organizations during the scrape
  * add members to an organization: in which case you must create/scrape and yield the organization, but you don't need to yield the person - instead you can call [add_member](https://github.com/opencivicdata/pupa/blob/master/pupa/scrape/popolo.py#L170) with a dictionary of any properties you know about the person (e.g. their name) and then those properties will be used to [resolve](https://github.com/opencivicdata/pupa/blob/master/pupa/importers/base.py#L117) the person

What you're doing is scraping multiple different forms of the same person, and then expecting Pupa to deduplicate the *scraped* objects. Pupa doesn't do that. It only reconciles scraped objects with saved objects (in the DB) to avoid the creation of duplicates.

**Comment by patcon** [comment]
OK, great! Any pointers at cities that scrape multiple sources would be super-helpful.

I'll stick with this approach of using my lookup dict in one scraper, since I've sorted out the tough parts and I'm excited to work on events now, but I'll pull them apart and do it properly before expecting a merge. I definitely won't leave this in perma-limbo, as I'd love it upstream eventually :)

**Pull request closed without merging**